### PR TITLE
Updating Algos.

### PR DESCRIPTION
**Pull request by SklCandy420**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by Panquesito7** [comment]
Rebase needed. Please [rebase](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github).

File: search/Linear Search.cpp [hunk 1]
```diff
@@ -1,47 +1,32 @@
...
-	}
-	else
-	{
-		cout << "\nNot found";
-	}
-
-	return 0;
-}
+#include <iostream> 
+using namespace std;
```
Comment (ayaankhan98) [inline search/Linear Search.cpp:4]: don't use ```using namespace std```
use ```std::xyz``` whereever necessary

File: search/Linear Search.cpp [hunk 2]
```diff
@@ -1,47 +1,32 @@
...
-	{
-		cout << "\nNot found";
-	}
-
-	return 0;
-}
+#include <iostream> 
+using namespace std; 
+
+int search(int arr[], int n, int x)
```
Comment (ayaankhan98) [inline search/Linear Search.cpp:6]: documentaion missing
follow proper doxygen documentation gudelines when writing documentation

File: search/Linear Search.cpp [hunk 3]
```diff
@@ -1,47 +1,32 @@
...
-	}
-
-	return 0;
-}
+#include <iostream> 
+using namespace std; 
+
+int search(int arr[], int n, int x) 
+{ 
+	int i;
```
Comment (ayaankhan98) [inline search/Linear Search.cpp:8]: this varibale can be made local to for loop

File: search/Linear Search.cpp [hunk 4]
```diff
@@ -1,47 +1,32 @@
...
+
+int search(int arr[], int n, int x) 
+{ 
+	int i; 
+	for (i = 0; i < n; i++) 
+		if (arr[i] == x) 
+			return i; 
+	return -1; 
+} 
+
```
Comment (ayaankhan98) [inline search/Linear Search.cpp:14]: add some tests.
write a function ```void tests()``` to test the algorithm

File: sorting/Bubble Sort.cpp [hunk 5]
```diff
@@ -1,55 +1,53 @@
-//Bubble Sort
+#include <bits/stdc++.h>
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:1]: dont't use ```bits/stdc++.h```

File: sorting/Bubble Sort.cpp [hunk 6]
```diff
@@ -1,55 +1,53 @@
-//Bubble Sort
+#include <bits/stdc++.h> 
+using namespace std;
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:2]: avoid using `using namespace std`

use ```std::xyz``` whereever necessary

File: sorting/Bubble Sort.cpp [hunk 7]
```diff
@@ -1,55 +1,53 @@
-//Bubble Sort
+#include <bits/stdc++.h> 
+using namespace std; 
 
-#include <iostream>
-#include <vector>
-using namespace std;
+void swap(int *xp, int *yp)
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:4]: no need of writing a seprate swap function you can use ```std::swap()```

**Comment by kvedala** [comment]
@ayaankhan98 this PR is quite outdated and needs to be rebased on the main branch as @Panquesito7 reviewed above. Until then, any comments and reviews would be futile.

File: sorting/Bubble Sort.cpp [hunk 8]
```diff
@@ -1,55 +1,53 @@
...
+void swap(int *xp, int *yp) 
+{ 
+	int temp = *xp; 
+	*xp = *yp; 
+	*yp = temp; 
+} 
 
-int main()
-{
+// A function to implement bubble sort
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:11]: please write proper function documentation
follow doxygen documentation guideliens while writing documentation

File: sorting/Bubble Sort.cpp [hunk 9]
```diff
@@ -1,55 +1,53 @@
...
+	*xp = *yp; 
+	*yp = temp; 
+} 
 
-int main()
-{
+// A function to implement bubble sort 
+void bubbleSort(int arr[], int n) 
+{ 
+	int i, j;
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:14]: these variables can be made local to their respective for loops

File: sorting/Bubble Sort.cpp [hunk 10]
```diff
@@ -1,55 +1,53 @@
...
+// A function to implement bubble sort 
+void bubbleSort(int arr[], int n) 
+{ 
+	int i, j; 
+	for (i = 0; i < n-1; i++)	 
+	
+	// Last i elements are already in place 
+	for (j = 0; j < n-i-1; j++) 
+		if (arr[j] > arr[j+1]) 
+			swap(&arr[j], &arr[j+1]);
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:20]: use ```std::swap()```

File: sorting/Bubble Sort.cpp [hunk 11]
```diff
@@ -1,55 +1,53 @@
...
+	// Last i elements are already in place 
+	for (j = 0; j < n-i-1; j++) 
+		if (arr[j] > arr[j+1]) 
+			swap(&arr[j], &arr[j+1]); 
+} 
+
+/* Function to print an array */
+void printArray(int arr[], int n) 
+{ 
+	int i;
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:26]: variable can be made local to loop

File: search/Linear Search.cpp [hunk 12]
```diff
@@ -1,47 +1,32 @@
...
+int main(void) 
+{ 
+	int n;
+	cin >> n;
```
Comment (ayaankhan98) [inline search/Linear Search.cpp:18]: use ```std::cin```

File: search/Linear Search.cpp [hunk 13]
```diff
@@ -1,47 +1,32 @@
...
+	
+	int x;
+	cin >> x;
```
Comment (ayaankhan98) [inline search/Linear Search.cpp:21]: use ```std::cin```

File: search/Linear Search.cpp [hunk 14]
```diff
@@ -1,47 +1,32 @@
...
+	
+	int arr[n];
+	for(int i=0; i<n; i++)
+		cin >> arr[i];
+	
+	int result = search(arr, n, x); 
+
+	(result == -1)? cout<<"Element is not present in array"
+				: cout<<"Element is present at index " <<result;
```
Comment (ayaankhan98) [inline search/Linear Search.cpp:30]: use ```std::cout```

File: sorting/Bubble Sort.cpp [hunk 15]
```diff
@@ -1,55 +1,53 @@
...
+		if (arr[j] > arr[j+1]) 
+			swap(&arr[j], &arr[j+1]); 
+} 
+
+/* Function to print an array */
+void printArray(int arr[], int n) 
+{ 
+	int i; 
+	for (i = 0; i < n; i++) 
+		cout << arr[i] << " "<<"\n";
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:28]: use ```std::cout```

File: sorting/Bubble Sort.cpp [hunk 16]
```diff
@@ -1,55 +1,53 @@
...
-	{
-		cin >> num;
-		numbers.push_back(num);
-	}
+	int arr[n];
+	
+	//looping for input of array
+	
+	for(int i =0; i< n; i++)
+		cin >> arr[i];
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:42]: use ```std::cin```

File: sorting/Bubble Sort.cpp [hunk 17]
```diff
@@ -1,55 +1,53 @@
...
+
+/* Function to print an array */
+void printArray(int arr[], int n) 
+{ 
+	int i; 
+	for (i = 0; i < n; i++) 
+		cout << arr[i] << " "<<"\n"; 
+	
+} 
+
```
Comment (ayaankhan98) [inline sorting/Bubble Sort.cpp:31]: add some test cases to test the algorithm

**Comment by ayaankhan98** [comment]
> @ayaankhan98 this PR is quite outdated and needs to be rebased on the main branch as @Panquesito7 reviewed above. Until then, any comments and reviews would be futile.

oh thanks, i didn't seen that

**Pull request closed without merging**